src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems

```diff
@@ -747,6 +747,7 @@
     <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\InteropServices\ICustomMarshaler.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\InteropServices\ICustomQueryInterface.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\InteropServices\IDynamicInterfaceCastable.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\InteropServices\IMarshal.cs" />
```

Comment:
You also need to move the linker directives in src\libraries\System.Runtime.InteropServices\src\ILLinkTrim.xml . Otherwise, this will be broken.

We do not seem to have any tests for this, so the fact that it is broken will be noticed only the change hits WinForms/WPF repos that depend on this type. Could you please get this fixed asap?